test-helpers/src/mpc_network/mocks.rs

```diff
@@ -79,47 +128,20 @@ impl MpcNetwork for MockNetwork {
     }
 
     async fn send_message(&mut self, message: NetworkOutbound) -> Result<(), MpcNetworkError> {
-        let msg_bytes = serde_json::to_vec(&message)
-            .map_err(|e| MpcNetworkError::SerializationError(e.to_string()))?;
-
-        // Write the message length first
-        self.mock_conn
-            .write_u64(msg_bytes.len() as u64)
-            .await
-            .map_err(|err| MpcNetworkError::SendError(err.to_string()))?;
-
-        // Write the byte buffer
-        self.mock_conn
-            .write_all(&msg_bytes)
-            .await
-            .map_err(|err| MpcNetworkError::SendError(err.to_string()))?;
-
+        self.mock_conn.send(message);
         Ok(())
     }
 
     async fn receive_message(&mut self) -> Result<NetworkOutbound, MpcNetworkError> {
-        // Receive the message length first
-        let msg_len = self
-            .mock_conn
-            .read_u64()
-            .await
-            .map_err(|err| MpcNetworkError::RecvError(err.to_string()))?;
-
-        let mut buf = vec![0u8; msg_len as usize];
-        self.mock_conn
-            .read_exact(&mut buf)
-            .await
-            .map_err(|err| MpcNetworkError::RecvError(err.to_string()))?;
-
-        let msg: NetworkOutbound = serde_json::from_slice(&buf)
-            .map_err(|e| MpcNetworkError::SerializationError(e.to_string()))?;
+        let msg = self.mock_conn.recv().await;
         Ok(msg)
     }
 
     async fn exchange_messages(
         &mut self,
         message: NetworkOutbound,
     ) -> Result<NetworkOutbound, MpcNetworkError> {
+        panic!("exchange called");
```

Comment:
might be clearer to use an `unimplemented!()` macro here & remove the code below if this function is never meant to be called

Reply:
Yeah oops, left that in for testing, the implementation here should be available, nice catch